File: 1154. Day of the Year/solution.py
class Solution:
    def dayOfYear(self, date: str) -> int:
        year, month, day = map(int, date.split('-'))
        days = [31,28,31,30,31,30,31,31,30,31,30,31]
        if year % 400 == 0 or (year % 4 == 0 and year % 100 != 0):
            days[1] = 29
        return sum(days[:month-1]) + day

File: 1154. Day of the Year/solution.js
/**
 * @param {string} date
 * @return {number}
 */
var dayOfYear = function(date) {
    let arr = date.split('-');
    let year = parseInt(arr[0]);
    let month = parseInt(arr[1]);
    let day = parseInt(arr[2]);
    let days = 0;
    let isLeapYear = false;
    if(year % 4 === 0 && year % 100 !== 0 || year % 400 === 0){
        isLeapYear = true;
    }
    for(let i = 1; i < month; i++){
        if(i === 2){
            if(isLeapYear){
                days += 29;
            }else{
                days += 28;
            }
        }else if(i === 4 || i === 6 || i === 9 || i === 11){
            days += 30;
        }else{
            days += 31;
        }
    }
    days += day;
    return days;
};